airflow/models/xcom.py

```diff
@@ -63,7 +63,7 @@ def init_on_load(self):
         i.e automatically deserialize Xcom value when loading from DB.
         """
         try:
-            self.value = XCom.deserialize_value(self)
+            self.value = XCom.orm_deserialize_value(self)
```

Comment:
Why not change this to `self.deserialize_value` value instead. This way it will use the `deserialize_value` from the CustomXCom Backend.

Reply:
@kaxil this will not help as custom backends can perform any logic in this function and return any type of object (that was initial issues #12315 ). So doing a query on XCom may cause tenths of requests :D That's why I wanted to decouple it.

Reply:
I think either way we should make this a `self.` call
```suggestion
            self.value = self.orm_deserialize_value()
```

Reply:
Done in 18513c2